lib/utils/logger.ts

```diff
@@ -0,0 +1,66 @@
+import { EOL as eol } from 'os';
+import _ = require('lodash');
+import chalk from 'chalk';
+import { StreamLogger } from 'resin-stream-logger';
+
+export class Logger {
+	public streams: {
+		build: NodeJS.ReadWriteStream;
+		info: NodeJS.ReadWriteStream;
+		debug: NodeJS.ReadWriteStream;
+		success: NodeJS.ReadWriteStream;
+		warn: NodeJS.ReadWriteStream;
+		error: NodeJS.ReadWriteStream;
+	};
+
+	public formatMessage: (name: string, message: string) => string;
+
+	constructor() {
+		const logger = new StreamLogger();
+		logger.addPrefix('build', chalk.blue('[Build]'));
+		logger.addPrefix('info', chalk.cyan('[Info]'));
+		logger.addPrefix('debug', chalk.magenta('[Debug]'));
+		logger.addPrefix('success', chalk.green('[Success]'));
+		logger.addPrefix('warn', chalk.yellow('[Warn]'));
+		logger.addPrefix('error', chalk.red('[Error]'));
+
+		this.streams = {
+			build: logger.createLogStream('build'),
+			info: logger.createLogStream('info'),
+			debug: logger.createLogStream('debug'),
+			success: logger.createLogStream('success'),
+			warn: logger.createLogStream('warn'),
+			error: logger.createLogStream('error')
+		};
+
+		_.mapKeys(this.streams, function(stream, key) {
```

Comment:
This should probably be `forEach` since we aren't doing anything with its results and also some `return`s could be removed as a result.